### PR TITLE
Build airflow 2.0 from Dockerfile for additional Python modules insta…

### PR DESCRIPTION
**Pull request by mclaurentiu**
…llation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,9 @@ build-volumes:
 	@ sudo mkdir -p  ./airflow2/logs ./airflow2/plugins ./airflow2/dags
 	@ sudo chmod 777 ./airflow2/logs ./airflow2/plugins ./airflow2/dags
 
+build-network:
+	@ docker network create -d bridge elk
```
Comment (costezki) [inline Makefile:8]: Why the default network is not sufficient? 
Do we want to use this network across docker-container files? And if so then perhaps we can call it something else.

File: Makefile [hunk 2]
```diff
@@ -37,6 +40,20 @@ stop-airflow2:
 	@ echo "$(BUILD_PRINT)Stopping the AirFlow services"
 	@ docker-compose --file ./airflow2/docker-compose.yml --env-file ../.env down
 
+start-tika: build-volumes start-storage
```
Comment (costezki) [inline Makefile:43]: Tika starts once the elk starts, not sure we need this make target.
If we need to control tika separately, I propose then to move it out of the elk docker-compose into a dedicated compose file.

File: Makefile [hunk 3]
```diff
@@ -37,6 +40,20 @@ stop-airflow2:
...
+	@ echo "$(BUILD_PRINT)Starting Apache Tika"
+	@ echo "$(BUILD_PRINT)Warning: the shared folders need R/W permissions"
+	@ docker-compose --file docker-compose.yml --env-file .env up -d apache_tika
+
+stop-tika:
+	@ echo "$(BUILD_PRINT)Stopping Apache Tika"
+	@ docker-compose --file docker-compose.yml --env-file .env down
+
+start-elk: build-volumes start-storage
```
Comment (costezki) [inline Makefile:52]: let's avoid specifying the services, in the command line. The reason for multiple compose files is to group the services that should start together. If we need a separation between tika and elk, we shall do so.

File: airflow2/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,4 @@
+FROM apache/airflow:2.0.1-python3.8
+
+USER airflow
+RUN pip install --no-cache-dir --user tika elasticsearch
```
Comment (costezki) [inline airflow2/Dockerfile:4]: we can add a few more dependencies, we know will be used in the future. 

`mlflow nympy pandas pytorch  scikit-learn spacy `

for a start

File: airflow2/docker-compose.yml [hunk 5]
```diff
@@ -39,7 +39,8 @@
 version: '3'
 x-airflow-common:
   &airflow-common
-  image: ${AIRFLOW_IMAGE_NAME:-apache/airflow:2.0.1-python3.8}
+  build:
+    context: .
```
Comment (costezki) [inline airflow2/docker-compose.yml:43]: this looks about right.

Also perhaps it is time for a little refactoring, moving the elk related containers into a subfolder.

File: airflow2/docker-compose.yml [hunk 6]
```diff
@@ -74,6 +75,9 @@ services:
       interval: 5s
       retries: 5
     restart: always
+    networks:
```
Comment (costezki) [inline airflow2/docker-compose.yml:78]: why do we need the elk and airflow services in the same network?

File: docker-compose.yml [hunk 7]
```diff
@@ -53,6 +55,7 @@ services:
       - elasticsearch
 
   kibana:
+    container_name: kibana
```
Comment (costezki) [inline docker-compose.yml:58]: I assume these names are necessary to be able to call services internally between each other. 
We can rely on the implicit docker-compose names for the containers. If we need explicit container names, lets set them only for the services that rely on the custom names.

**Review by costezki — APPROVED** [review]
nice extension

File: Makefile (hunk 2 above)
Comment (mclaurentiu) [reply Makefile:43]: Tika does not start when ELK starts.

File: Makefile (hunk 3 above)
Comment (mclaurentiu) [reply Makefile:52]: I've removed the Makefile code.

File: airflow2/docker-compose.yml (hunk 5 above)
Comment (mclaurentiu) [reply airflow2/docker-compose.yml:43]: I agree. Please see https://meaningfy.atlassian.net/browse/SC-68

**Pull request merged**